### ISSUE TITLE
Ownership of products after re-install (Android Sandbox)

### ISSUE THREAD

**Issue by bwalsh** [j3k0/cordova-plugin-purchase#243]
Thanks for a great plug-in.

Just a quick to understand expected behavior.

`"As a user of the app, when I want to unlock content I want to buy the content trough the android store"`
This works fine, without a hitch.

`"As a returning user to the app, when I re-install the app, I expect my purchased content to be unlocked "`
On re-install of the app, the final state of the product, after update callback is:
state:valid, owned:false.
I expected owned to be true.  Am I missing something?

**Comment by aesculus** [comment]
Mine resets on:
<code>store.when("product").updated(function (product) {</code>

It seems like it takes a couple of passes though. When first connected it comes back as false but after a few seconds I get another firing and it is set to true.

**Comment by aesculus** [comment]
A bit off topic. Apple rejected my app because I did not include a Restore Purchase button. I told them that I automatically detect purchases when the app starts and make sure the user has access to the content they paid for no matter what the device status is.

They said they understood my logic but the rules are that you MUST have a Restore button :-(

**Comment by aesculus** [comment]
You have to call store.refresh() again in your restore button. It seems to require this after the initial call. This is loosely documented.

**Comment by fer8a** [comment]
@aesculus creating a visible button to 'restore' purchases in your app that calls `store.refresh()` when is clicked did solve the issue with Apple? They did the same to my app.

**Comment by aesculus** [comment]
Yes. Even though it was not needed because store.refresh() automatically does a restore, they still wanted a button.  And they admitted in my app it did not do anything. :-)

**Comment by mifkys** [comment]
Hello. Same problem. After app re-install subscription has state valid: true, but owned: false. All listeners on updated, approved, verified exists. What need to do to fix this issue?
